HPCC-18691 Do not challenge OPTIONS requests

rpastrana opened the pull request:
- Processes CORS OPTIONS requests without athorization

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@afishbeck please take a look. This resolves the tableau desktop communication error. We can discuss if we feel this opens a security issue.

richardkchapman commented:
Typo in commit message.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 170795ac1a9b222e33c78fe0179bc1d0c44ad1e7
Empty table
eclwatch/DynamicESDLMethodWidget.js
eclwatch/LFDetailsWidget.js

Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 75 | 43.368 seconds |


Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |117 |117 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |745 |745 |0 |
|test (thor) |640 |640 |0 |
|test (roxie) |773 |773 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana committed:
HPCC-18691 Do not challenge OPTIONS requests

- Processess CORS OPTIONS requests without authorization

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 7f65516a2b0634e22ce3b9cfb4bcfa536f97c372
Empty table
eclwatch/DynamicESDLMethodWidget.js
eclwatch/LFDetailsWidget.js

Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 75 | 40.873 seconds |


Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |117 |117 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |745 |745 |0 |
|test (thor) |640 |640 |0 |
|test (roxie) |773 |773 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

afishbeck commented on esp/bindings/http/platform/httpservice.cpp:
```diff
@@ -259,6 +259,9 @@ int CEspHttpServer::processRequest()
         ESPLOG(LogMin, "%s %s, from %s@%s", method.str(), m_request->getPath(pathStr).str(), (userid) ? userid : "unknown", m_request->getPeer(peerStr).str());
 
+        if(strieq(method.str(), OPTIONS_METHOD))
+            return onOptions();
```
Should probably delete the identical lines below?

afishbeck commented:
@rpastrana looks good except duplicate code.  But I wonder if this was always like this, if so this change should probably be 6.4.6.  Unless it was broken by the session changes that are targeting 7.0.

rpastrana committed:
HPCC-18691 Do not challenge OPTIONS requests

- Processess CORS OPTIONS requests without authorization
- Removes duplicate lines

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :x:
Sha: ca6db13186ae5e77d9d82af08516ca59ce9b9a77
Empty table
eclwatch/DynamicESDLMethodWidget.js
eclwatch/LFDetailsWidget.js

Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 75 | 42.186 seconds |


Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |117 |117 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |745 |745 |0 |
|test (thor) |640 |639 |1 |
|test (roxie) |773 |773 |0 |


[Errors:](https://gist.github.com/6dba7ed800225d6aff1575ac26c71c8e)
- test
  - thor
    - 506\. [sortnorm (algo=parquicksort)](https://gist.github.com/HPCCSmoketest/6dba7ed800225d6aff1575ac26c71c8e#file-sortnorm-algo-parquicksort-w20171108-125439-2-log), 

HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented:
new pull request targets earlier branch: 
https://github.com/hpcc-systems/HPCC-Platform/pull/10628
